### PR TITLE
forgerock: fix handling of idm_core object payloads

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

See title.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

File: packages/forgerock/data_stream/idm_core/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -29,6 +29,11 @@ processors:
       field: event.reason
       copy_from: forgerock.payload
       ignore_failure: true
+      if: ctx.forgerock?.payload instanceof String
+  - set:
+      field: event.reason
+      copy_from: forgerock.payload.message
+      ignore_failure: true
```
Comment (kcreddy) [inline packages/forgerock/data_stream/idm_core/elasticsearch/ingest_pipeline/default.yml:36]:
```suggestion
      if: ctx.forgerock?.payload instanceof Map
      ignore_empty_value: true
```
?

File: packages/forgerock/data_stream/idm_core/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -37,11 +42,22 @@ processors:
...
+      if: ctx.forgerock?.payload instanceof String && (ctx.tags == null || !ctx.tags.contains('preserve_duplicate_custom_fields'))
       field:
         - forgerock.payload
       ignore_failure: true
       ignore_missing: true
+  - remove:
+      if: ctx.forgerock?.payload instanceof Map && (ctx.tags == null || !ctx.tags.contains('preserve_duplicate_custom_fields'))
+      field:
+        - forgerock.payload.message
+      ignore_failure: true
```
Comment (kcreddy) [inline packages/forgerock/data_stream/idm_core/elasticsearch/ingest_pipeline/default.yml:54]: Do we need `ignore_failure: true` in here? 
Same for below.

**Review by kcreddy — APPROVED** [review]
Minor clarifications. LGTM otherwise

File: packages/forgerock/data_stream/idm_core/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (efd6) [reply packages/forgerock/data_stream/idm_core/elasticsearch/ingest_pipeline/default.yml:54]: ...and above. I'll remove them.

File: packages/forgerock/data_stream/idm_core/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (efd6) [reply packages/forgerock/data_stream/idm_core/elasticsearch/ingest_pipeline/default.yml:36]: Do we need the condition?

Comment (kcreddy) [reply packages/forgerock/data_stream/idm_core/elasticsearch/ingest_pipeline/default.yml]: We don't 👍🏼

**Review by kcreddy — APPROVED** [review]

**Pull request merged**